libbeat/common/capabilities_linux.go

```diff
@@ -50,12 +51,12 @@ func (c Capabilities) Check(set []string) bool {
 func GetCapabilities() (Capabilities, error) {
 	p, err := sysinfo.Self()
 	if err != nil {
-		return Capabilities{}, errors.Wrap(err, "failed to read self process information")
+		return Capabilities{}, fmt.Errorf("failed to read self process information: %w", err)
 	}
 
 	if c, ok := p.(types.Capabilities); ok {
 		capabilities, err := c.Capabilities()
-		return Capabilities(*capabilities), errors.Wrap(err, "failed to read process capabilities")
+		return Capabilities(*capabilities), fmt.Errorf("failed to read process capabilities: %w", err)
```

Comment:
This is calling through to go-sysinfo: https://github.com/search?q=repo%3Aelastic%2Fgo-sysinfo%20Capabilities()&type=code

It looks like there is only one implementation of `Capabilities` for process and there is a path where it returns `nil`, so `*capabilities` isn't guaranteed to be safe.

This function only looks to be used in one place:

https://github.com/elastic/beats/blob/169fe5864ad3afaf3a60a8a91042a94315c8d78c/metricbeat/helper/socket/ptable_linux.go#L30-L37

You should be able to change the return type to `*Capabilities` and then return `nil` instead of `Capabilties(*capabilities)` instead without issue and eliminate a possible `nil` dereference.